lib/ls.js

```diff
@@ -146,7 +146,8 @@ function getLite (data, noname) {
     lite.problems.push('extraneous: ' + getPackageId(data) + ' ' + (data.path || ''))
   }
 
-  if (data.error && data.path !== path.resolve(npm.globalDir, '..')) {
+  if (data.error && data.path !== path.resolve(npm.globalDir, '..') &&
+      (data.error.code !== 'ENOENT' || noname)) {
```

Comment:
add test that DOES trigger an error at the bottom from a top level package. Invalid package.json.